### PR TITLE
Added Apex Code Snippets

### PR DESCRIPTION
**Pull request by ntotten**
### What does this PR do?
Adds code snippets to the apex extension.

### What issues does this PR fix or reference?
N/A

### CODE REVIEW

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            "for (${1:variable} : [${2:SOQL}]) {",
+            "    $0",
+            "}"
+        ],
+        "description": "for loop"
+    },
+    "Reverse for loop": {
+        "prefix": "forr",
+        "body": [
+            "for (Integer ${1:i} = ${2:length} - 1; ${1:i} >= 0 ; ${1:i}--)",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: This formatting is weird. Why is the `{` on the next line vs how we did it for the other `for` loops?

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            "{",
+            "    $0",
+            "}"
+        ],
+        "description": "Reverse for loop"
+    },
+    "for loop": {
+        "prefix": "fori",
+        "body": [
+            "for (Integer ${1:i} = 0; ${1:i} < ${2:length}; ${1:i}++)",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: Ditto.

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            "{",
+            "    $0",
+            "}"
+        ],
+        "description": "indexed for loop"
+    },
+    "if statement": {
+        "prefix": "if",
+        "body": [
+            "if (${1:true})",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: Ditto.

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            "{",
+            "    $0",
+            "}"
+        ],
+        "description": "if statement"
+    },
+    "Else statement": {
+        "prefix": "else",
+        "body": [
+            "else",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: Ditto.

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            "    $0",
+            "}"
+        ],
+        "description": "Else statement"
+    },
+    "Try finally": {
+        "prefix": "tryf",
+        "body": [
+            "try",
+            "{",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: Ditto.

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+            "    $0",
+            "}"
+        ],
+        "description": "Try finally"
+    },
+    "Try catch": {
+        "prefix": "try",
+        "body": [
+            "try",
+            "{",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: Ditto.

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            "    $0",
+            "}"
+        ],
+        "description": "Try catch"
+    },
+    "Try catch finally": {
+        "prefix": "trycf",
+        "body": [
+            "try",
+            "{",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: Ditto.

File: packages/salesforcedx-vscode-apex/snippets/apex.json [hunk 8]
```diff
@@ -0,0 +1,176 @@
+{
+    "Class": {
+        "prefix": "class",
+        "body": [
+            "class ${1:Name}",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/snippets/apex.json]: Do we need to have a modifier? I'm not sure if the code will compile without a default modifier like public|global

**Comment by ntotten** [comment]
Updated per comments.

**Comment by vazexqi** [comment]
@DatGuyJonathan - This looks low risk to me. Since it's going into the develop branch this shouldn't affect anything on the v43.2.0 release this Saturday. Any objections?

**Comment by vazexqi** [comment]
"Our workspace recommends Prettier extension. Is there a way to exclude this file? I like the format of this file as-is, but Prettier is going to reformat it."<-- Any harm with just formatting it with Prettier? Are you worried about the spacing/indentation that could happen? We can add it to a .prettierignore to ignore formatting it. I _think_ this will also prevent VS Code's Prettier extension from formatting it.

**Comment by ntotten** [comment]
#2 is cool. I didn’t know this was an option. I can update in the morning.

#3 I think there are a lot more we can add. This is really just the first and most obvious batch. I’ll keep adding more and ask for community feedback.

**Comment by vazexqi** [comment]
I'm taking control of this PR and will get it merged in. I will bring it from the external branch that Nate has into this repo.

**Comment by vazexqi** [comment]
@ntotten could you grant me permission to modify your branch per https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

**Comment by ntotten** [comment]
@vazexqi It shows as being checked already.

**Comment by vazexqi** [comment]
snippets

**Comment by ntotten** [comment]
No problem. I think I copied it from our docs, but I must have messed it up.

**Pull request merged**